pkgs/development/libraries/libglvnd/default.nix

```diff
@@ -31,8 +31,11 @@ stdenv.mkDerivation rec {
     "-Wno-error=array-bounds"
   ] ++ lib.optional stdenv.cc.isClang "-Wno-error");
 
-  # Indirectly: https://bugs.freedesktop.org/show_bug.cgi?id=35268
-  configureFlags  = lib.optional stdenv.hostPlatform.isMusl "--disable-tls";
+  configureFlags  = []
+    # Indirectly: https://bugs.freedesktop.org/show_bug.cgi?id=35268
+    ++ lib.optional stdenv.hostPlatform.isMusl "--disable-tls"
+    # Remove when aarch64-darwin asm support is upstream
+    ++ lib.optional (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64) "--disable-asm";
```

Comment:
```suggestion
    # Indirectly: https://bugs.freedesktop.org/show_bug.cgi?id=35268
  configureFlags  = lib.optional stdenv.hostPlatform.isMusl "--disable-tls"
    # Remove when aarch64-darwin asm support is upstream
    ++ lib.optional (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64) "--disable-asm";
```

Reply:
Yeah, I wasn't sure either, since the first comment is misaligned in that case.

Reply:
Is there an upstream issue we could link? After that I think we can merge this.

Reply:
There doesn't seem to be one yet. From the brief description it sounds like it fails at runtime due to invalid instructions - could you maybe also provide that error message?

`libglvnd` has code for AArch64 that should probably work for the Apple M1 SoC as well so this runtime error might be due to an incorrect host CPU detection (https://github.com/NVIDIA/libglvnd/blob/acc654454867c7cdd681cc1f60f858bcd6e5e729/configure.ac#L119).

Anyway, I'd be fine with `--disable-asm` for `aarch64-darwin` but finding a proper solution / contacting upstream (they might not be that "enthusiastic" about Apple stuff but in this case it likely doesn't matter) would of course be best.

Reply:
Added a link to the upstream issue. This is a compile time error, not a run time one.

Let me know if there's anything else blocking this.

Reply:
Rebased and updated the link to point to gitlab after c6f1dc2320db5b28136ca4a78ffce5cf0b7cdba5.

Reply:
Thanks! LGTM then :)

> This is a compile time error, not a run time one.

Interesting, that might actually be an issue with `Apple clang` then (from https://gitlab.freedesktop.org/glvnd/libglvnd/-/issues/216 it looks like the compiler cannot (properly) handle inline assembly but the output could be misleading).